code/code/cmd/cmd_deathstroke.cc

```diff
@@ -62,69 +62,113 @@ static int deathstroke(TBeing *caster, TBeing *victim)
 
   int dam = caster->getSkillDam(victim, SKILL_DEATHSTROKE, lev, caster->getAdvLearning(SKILL_DEATHSTROKE));
 
-  if ((caster->bSuccess(bKnown+ percent, SKILL_DEATHSTROKE) &&
-         (i = caster->specialAttack(victim,SKILL_DEATHSTROKE)) && 
```

Comment:
What's in specialAttack() here that you are removing?

Reply:
I removed it just because I wasn't really sure why all that was added to the skill check in the first place.  Honestly the initial code for the skill check was pretty convoluted (just my opinion) shifting bits in the algorithm and having not just one success check but multiple that had to succeed, which is part of the reason I suspect why it failed frequently in game.  My goal was to simplify the algorithm but to keep it tied to armor and to bring in some stats as well (including the victim's stats).

Reply:
Just to jump in here in hopes of helping a bit, as a lurker who wishes I had more spare time so I could contribute to the code, I dug into the specialAttack method a while back to see how it works. For all intents and purposes it's basically a comparison of attacker's offense vs defender's defense, but the formulas for calculating those offense and defense values rely heavily on level. 

Because of that reliance on level, there's a certain point where it's really difficult for a PC to increase their offense value any further, so specialAttack essentially ends up functioning as kind of a convoluted way to greatly reduce the chance physical PC special attacks (think backstab, deathstroke, etc) have of landing successfully on mobs 10-20+ levels higher (depending on PC offense). At around 20 levels higher the chance of the specialAttack check being successful becomes almost nil, usually.

So, without the call to specialAttack() here, what I believe will likely happen is that if the PC passes the bSuccess check (which is really just a check against their learnedness in the skill as a value from 0-100 where 100 = maxed/maxed, modified  +/- 25% by focus, and now I believe the difficulty check that was added back in) they'll be able to land deathstroke against a mob of any level without any test of offense vs defense.

Hopefully this helps. I still have the spreadsheet somewhere that maps out and tests the function so I can try to dig it up if you want more detail.

Reply:
Thanks @drewhershey I thought these were going through some hit check. Let me dig into this.

There should be some check for skill success AND there should be some check that compares user VS victim offense and defense. 

So for this @parkscs I would add that back in and let's start work (in a new branch) on an adjustment to how special attack hits are calculated. I know it sucks that this is hard to land but the problem is bigger than just death stroke and let's go ahead and adjust it for everything.

---

Here's some brainstorming thoughts on this:

skill success:
- skill success chance is based on learnedness and modified by the difficulty of the skill, focus and karma.
- should not be modified by other caster or victim stats or other things
- Is just essentially, do I know how to do this move?

hitting:
- offense takes into account level, skills, position, dex, equipment etc
- defense takes into account level, armor, skills, position
- So those two things really account for all factors. We shouldn't need to factor anything else.
- So it's just a matter of how we compare those two values.

However, I know that it is stupid hard to hit a much higher level mob. We need to adjust this formula so that higher level mobs aren't un-hittable but also still present a greater challenge.

Reply:
Whoops, just realized I posted on my work account before. I wrote up an issue about this exact topic a few years back actually:

https://github.com/sneezymud/sneezymud/issues/163

Like I mention in that issue, it looks like over the years coders started confusing bSuccess checks (e.g. the first category @wershlak mentioned - whether the PC knows HOW to do something), the secondary check (referred to as "hitting") that determines if the attack succeeds based on factors out of the PC's control (weather, vision, level, whatever), and then a third kind of check that I think is referred to in the code as a "hard fail" check - basically an instafail scenario, like using a blunt weapon to backstab (although one could argue hard fails lead to less dynamic gameplay, rather than letting players deal with the consequences of their actions?).

Also, and you probably already know this, but, for most active skills (including prayers and spells, I'm pretty sure) there's two separate functions - <name> and "do<name>". So "backstab" and "doBackstab" for instance. The "do" function appears to be intended to contain all the hard fail checks, where a condition being true just returns FALSE immediately, or otherwise calls the "non-do" function if all the hard fail checks pass. Yet in many skills those types of hard fail checks have started getting placed in the "non-do" function.

For example: there's a check in doBackstab() to fail if the victim isn't found in the room - which makes sense. But then in backstab(), which gets called by doBackstab(), there's a hard fail if the thief is in a peaceful room. Why isn't that just in doBackstab() with the other hard fails?

It makes things confusing and introduces a lot of room for bugs IMO. So it could make sense to try to improve that separation of concerns at the same time, since it's related.

But I also very much agree that the physical special attack success formula could use some tweaks, based on my memories of hundreds of missed backstabs by my thief during high level groups over the years...

Oh, and if you want to see an example of how powerful a skill becomes when you remove the specialAttack() check, take a look at garrotte. It's super strong against high level casters because it only requires passing a bSuccess check to land.